src/utils/src/url.js

```diff
@@ -9,6 +9,7 @@ import { isDefined } from './object';
 
 /**
  * @private
+ * Adapted from https://github.com/lakenen/node-append-query, to fit a non-node runtime.
  */
 function serialize(obj, options = {}, prefix) {
```

Comment:
Why serialize manually, if qs library can do this? Is there something custom here? We could still delegate some of it to qs, no?

And isn't it better to only encode URLs in one place, if that's even possible?